snapcraft/internal/errors.py

```diff
@@ -250,6 +250,14 @@ def __init__(self, message):
         super().__init__(message=message)
 
 
+class PluginBaseError(SnapcraftError):
+
+    fmt = "The {plugin_name!r} plugin does not support snaps targeting base {base!r}."
```

Comment:
Perhaps "using" is a better word here, what do you think? When I think "target" I think "what do I run on?" which is typically Windows, Mac OS, or Linux, or perhaps Ubuntu, or even specific versions of Ubuntu. Specifying a base isn't really the same thing: nothing "runs on" core18, in fact, by virtue of being a snap it runs in a number of places and just happens to use core18 to get that functionality.